java/test/com/github/bazel_contrib/contrib_rules_jvm/junit5/BazelJUnitOutputListenerTest.java

```diff
@@ -37,11 +48,72 @@
 import org.xml.sax.InputSource;
 import org.xml.sax.SAXException;
 
-public class BazelJUnitOuputListenerTest {
+public class BazelJUnitOutputListenerTest {
 
   private TestDescriptor testDescriptor = new StubbedTestDescriptor(createId("descriptors"));
   private TestIdentifier identifier = TestIdentifier.from(testDescriptor);
 
+  /** This latch is used in TestAfterAllFails for testAfterAllFailuresAreReported */
+  private static final AtomicBoolean causeFailure = new AtomicBoolean(false);
+
+  static final class TestAfterAllFails {
+
+    @SuppressFBWarnings(value = "THROWS_METHOD_THROWS_RUNTIMEEXCEPTION")
+    @AfterAll
+    static void afterAll() {
+      if (causeFailure.get()) {
+        throw new RuntimeException("I always fail.");
+      }
+    }
+
+    @Test
+    void test() {}
+  }
+
+  @Test
+  public void testAfterAllFailuresAreReported() throws IOException {
+    causeFailure.set(true);
```

Comment:
Do you mean instead of my integration tests?
The other tests look like they pre-can create a `TestData` object and validate what the output looks like;

The bug I sought to fix was that the creation of the `TestData` object to begin with in the output listener was incorrect (missing the `@AfterAll`); I think there's value in the system test.

I can **also** add the unit test if you'd like.